variant/validators/amino_acid_deletion.py

```diff
@@ -0,0 +1,238 @@
+                else:
+                    hgvs_expr = self.get_hgvs_expr(classification, t, s, False)
+                    allele = self.get_allele_from_hgvs(hgvs_expr, errors)
+
+                if not allele:
+                    errors.append("Unable to find allele.")
+                else:
+                    # MANE Select Transcript
+                    if hgvs_expr not in mane_transcripts_dict.keys():
+                        mane_transcripts_dict[hgvs_expr] = {
+                            'classification_token': s,
+                            'transcript_token': t
+                        }
+                    if len(allele['state']['sequence']) == 3:
+                        allele['state']['sequence'] = \
+                            self._amino_acid_cache.convert_three_to_one(
+                                allele['state']['sequence'])
+
+                # Check ref start/end protein matches expected
+                if not errors:
+                    self.check_ref_aa(t, s.start_aa_del,
+                                      s.start_pos_del, errors)
+
+                    if not errors and s.end_aa_del:
+                        self.check_ref_aa(t, s.end_aa_del,
+                                          s.end_pos_del, errors)
+
+                if not errors and allele not in valid_alleles:
+                    results.append(self.get_validation_result(
+                        classification, True, 1, allele,
+                        self.human_description(t, s),
+                        self.concise_description(t, s), [], gene_tokens))
+
+                    valid_alleles.append(allele)
+                else:
+                    results.append(self.get_validation_result(
+                        classification, False, 1, allele,
+                        self.human_description(t, s),
+                        self.concise_description(t, s), errors, gene_tokens))
+
+        # Now add MANE transcripts to result
+        self.add_mane_transcript(classification, results, gene_tokens,
+                                 mane_transcripts_dict)
+
+    def check_ref_aa(self, t, aa, pos, errors):
+        """Check that reference amino acid matches actual amino acid.
+
+        :param string t: Transcript
+        :param str aa: Expected Amino Acid
+        :param str pos: Expected position
+        :param list errors: List of errors
+        """
+        ref_aa_del = \
+            self.seqrepo_access.sequence_at_position(t, pos)
+
+        if ref_aa_del and len(ref_aa_del) == 1 \
+                and len(aa) == 3:
+            ref_aa_del = \
+                self._amino_acid_cache.amino_acid_code_conversion[
+                    ref_aa_del]  # noqa: E501
+
+        if ref_aa_del != aa:
+            errors.append(f"Needed to find {aa} at "
+                          f"position {pos} on {t} "
+                          f"but found {ref_aa_del}")
+
+    def get_hgvs_expr(self, classification, t, s, is_hgvs) -> str:
```

Comment:
return type should be something like Tuple[str, bool]?

Reply:
Not for this class. Updated docs.